feat(sessions): Add option to filter by proj slug in query params [INGEST-1210]

iker-barriocanal opened the pull request:
Dashboards require the ability to filter by projects in query parameters, and right now this is only possible by the top-level `project` parameter. This PR introduces the feature to filter project slug and environment in the query. Filtering by the top-level parameter and filtering in the query at the same time is also supported.

iker-barriocanal committed:
Support filtering by proj slug in query

iker-barriocanal committed:
Add tests for project filtering

iker-barriocanal committed:
Add test for environment filtering

jjbayer reviewed (approved):
Two minor comments, otherwise this looks good!

iker-barriocanal committed:
Fix column warning

iker-barriocanal committed:
Use fixture instead of inheritance for consistency

Instead of inheriting from `TestCase` and `SnubaTestCase`, with the goal
of being consistent with the unit tests in the rest of the file, I've
refactored the code to use the `pytest.mark.django_db` fixture to use
the project. This refactor also keeps the code cleaner and easier to
read.

iker-barriocanal commented on src/sentry/snuba/metrics/query_builder.py:
```diff
@@ -160,6 +160,8 @@ def resolve_tags(org_id: int, input_: Any) -> Any:
         )
     if isinstance(input_, Column):
+        if input_.name == "project_id":
+            return input_
```
Reworded the comment instead of deleting it, since `Column("metric_id")` is not supported; see [802cebe](https://github.com/getsentry/sentry/pull/34281/commits/802cebedaeb811b172f813ecd20399a3daca19c9).
